Generating TLS rules with "go generate" fails

### Summary 

Following [the directions in the README](https://github.com/GoASTScanner/gas#generate-tls-rule) for generating up-to-date TLS rules do not seem to work for me:

- If I follow the instructions, I see warnings during the generate step, and the build itself fails
- If I remove `rules/tls.go`, then do `go generate`, then re-add `rules/tls.go`, I can build gas with up-to-date TLS rules

### Steps to reproduce the behavior

I've confirmed this in a docker container (version 1.10 of go), which has a clean environment:

```
$ docker run -it golang:1 /bin/sh
# go version
go version go1.10 linux/amd64
# go get github.com/GoASTScanner/gas/cmd/tlsconfig/...
# cd /go/src/github.com/GoASTScanner/gas
# go generate ./...
2018/03/09 22:23:46 Warning: cannot map cipher 'ECDHE-ECDSA-CHACHA20-POLY1305'
2018/03/09 22:23:46 Warning: cannot map cipher 'ECDHE-RSA-CHACHA20-POLY1305'
2018/03/09 22:23:46 Warning: cannot map cipher 'ECDHE-ECDSA-CHACHA20-POLY1305'
2018/03/09 22:23:46 Warning: cannot map cipher 'ECDHE-RSA-CHACHA20-POLY1305'
2018/03/09 22:23:46 Warning: cannot map cipher 'ECDHE-ECDSA-CHACHA20-POLY1305'
2018/03/09 22:23:46 Warning: cannot map cipher 'ECDHE-RSA-CHACHA20-POLY1305'
2018/03/09 22:23:46 Warning: cannot map cipher 'DHE-RSA-CHACHA20-POLY1305'
```

If I try to force a build by removing the gas binary and re-running "go get", I get a slightly different error:

```
# rm /go/bin/gas
# go get github.com/GoASTScanner/gas/...
# github.com/GoASTScanner/gas/rules
rules/rulelist.go:85:52: cannot use NewIntermediateTLSCheck (type func(gas.Config) (gas.Rule, []ast.Node)) as type gas.RuleBuilder in field value
# /go/bin/gas
/bin/sh: 13: /go/bin/gas: not found
```

In order to successfully build gas with the generated rules, here's the procedure I have to follow:

```
go get -d github.com/GoASTScanner/gas/...
go get github.com/GoASTScanner/gas/cmd/tlsconfig/...
cd /go/src/github.com/GoASTScanner/gas
rm rules/tls.go
go generate ./...
git checkout rules/tls.go
go get github.com/GoASTScanner/gas/...
```

(Probably not the most efficient way...I'm a bit of a go newbie.)

### Gas version

Current HEAD (7116c4d3a1bf0a6fac3bcc6e6914f9affb9b9883)

### Go version (output of 'go version')

go version go1.10 linux/amd64

### Operating system / Environment

- Docker container (linux, amd64)

### Expected behavior

Successful build of gas, with up-to-date TLS rules, following the instructions in the README

### Actual behavior

Have to remove `rules/tls.go`, then do `go generate`, then re-add `rules/tls.go` in order to build gas with up-to-date TLS rules

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -10,7 +10,7 @@
   branch = "master"
   name = "github.com/mozilla/tls-observatory"
   packages = ["constants"]
-  revision = "26cf7187f27c96815d0b4d4ad6da36c7161d4bf8"
+  revision = "8791a200eb40f8625a152bfb8336171305f5f35c"
 
 [[projects]]
   name = "github.com/nbutton23/zxcvbn-go"

--- a/rules/tls_config.go
+++ b/rules/tls_config.go
@@ -17,6 +17,8 @@ func NewModernTLSCheck(id string, conf gas.Config) (gas.Rule, []ast.Node) {
 		goodCiphers: []string{
 			"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
 			"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+			"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+			"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
 			"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
 			"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
 			"TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384",
@@ -36,6 +38,8 @@ func NewIntermediateTLSCheck(id string, conf gas.Config) (gas.Rule, []ast.Node) 
 		MinVersion:   0x0301,
 		MaxVersion:   0x0303,
 		goodCiphers: []string{
+			"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+			"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
 			"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
 			"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
 			"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
@@ -77,6 +81,8 @@ func NewOldTLSCheck(id string, conf gas.Config) (gas.Rule, []ast.Node) {
 		MinVersion:   0x0300,
 		MaxVersion:   0x0303,
 		goodCiphers: []string{
+			"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
+			"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
 			"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
 			"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
 			"TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",

